### ISSUE TITLE
Error when using scikit-learn 0.24

### ISSUE THREAD

**Issue by maxnoe** [jpmml/sklearn2pmml#261]
With sklearn 0.24, I get the following error with this minimal example:

```python
from sklearn.ensemble import RandomForestClassifier
import numpy as np
from sklearn2pmml.pipeline import PMMLPipeline


rng = np.random.default_rng()
X = rng.normal(size=(100, 3))
y = rng.integers(0, 2, size=100)

clf = RandomForestClassifier(n_estimators=10)
clf.fit(X, y)

pipeline = PMMLPipeline([
    ('model', clf)
])

from sklearn2pmml import sklearn2pmml
sklearn2pmml(pipeline, 'test.pmml')
```

Error:
```Traceback (most recent call last):
  File "test_pmml.py", line 14, in <module>
    ('model', clf)
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn2pmml/pipeline/__init__.py", line 48, in __init__
    super(PMMLPipeline, self).__init__(steps = steps)
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/utils/validation.py", line 63, in inner_f
    return f(*args, **kwargs)
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/pipeline.py", line 118, in __init__
    self._validate_steps()
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/pipeline.py", line 157, in _validate_steps
    self._validate_names(names)
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/utils/metaestimators.py", line 70, in _validate_names
    invalid_names = set(names).intersection(self.get_params(deep=False))
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/pipeline.py", line 137, in get_params
    return self._get_params('steps', deep=deep)
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/utils/metaestimators.py", line 29, in _get_params
    out = super().get_params(deep=deep)
  File "/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/base.py", line 195, in get_params
    value = getattr(self, key)
AttributeError: 'PMMLPipeline' object has no attribute 'apply_transformer'

```

With 0.22 and 0.23, it generates the following warning:

```
/home/maxnoe/Uni/FACT/aict-tools/venv/lib/python3.7/site-packages/sklearn/base.py:213: FutureWarning: From version 0.24, get_params will raise an AttributeError if a parameter cannot be retrieved as an instance attribute. Previously it would return None.
  FutureWarning)
```

With 0.21 and , it just works.

**Comment by vruusmann** [comment]
Was fixed in https://github.com/jpmml/sklearn2pmml/commit/d5c14b2133e9760b3179822bf1d718e472e0624c

Looks like Scikit-Learn people messed up the step validation logic between 0.23 and 0.24, so that all `sklearn.pipeline.Pipeline` subclasses (ex. `sklearn2pmml.pipeline.PMMLPipeline`) got broken.

TLDR: Update the SkLearn2PMML package (everytime you update the base Scikit-Learn package).

**Comment by maxnoe** [comment]
I have just installed this today. Is there not yet a release on pypi with that commit?

**Comment by vruusmann** [comment]
This fix is included in SkLearn2PMML 0.65.0 and newer. PyPI currently provides 0.66.0, which should be fine.

What's your SkLearn2PMML version?

**Comment by maxnoe** [comment]
Sorry, there was a `< 0.57` due to an old now fixed bug in the requirements, with the current version, all is fine.
